### PR TITLE
adding method to create set from keys of map of type comparable

### PR DESCRIPTION
**Pull request by jay-babu**
I added `github.com/google/gofuzz` to do fuzzy testing. If you are against it, then I will remove the dependency and hardcode the test input, but that makes the testing less resilient.

I decided against adding a method to create a `Set` from values of Map. Happy to do so if you want

### CODE REVIEW

**Comment by jayp0521** [comment]
https://github.com/deckarep/golang-set/issues/90

**Comment by deckarep** [comment]
So, I am going to lean towards not relying on the fuzz library because we’re dealing with types purely designed within the Go language.

I would definitely be in support of GoFuzz if this were something along the lines of a protocol where fuzzing is definitely warranted to ensure proper parsing of inputs.

**Comment by jayp0521** [comment]
@deckarep should be good to go

**Comment by jayp0521** [comment]
@deckarep any update?

**Comment by jay-babu** [comment]
@deckarep gentle reminder

File: set.go [hunk 1]
```diff
@@ -194,3 +194,27 @@ func NewThreadUnsafeSet[T comparable](vals ...T) Set[T] {
...
+	s := NewSet[T]()
+
+	for k := range val {
+		s.Add(k)
+	}
+
+	return s
+}
+
+// Creates and returns a new set with the given keys of the map.
```
Comment (jeffwidman) [inline set.go:210]: Per `godoc` convertions, these docstrings should start with the name of the func being documented.

Related:
* https://github.com/deckarep/golang-set/pull/106

Comment (deckarep) [reply set.go:210]: @jeffwidman - your keen eye strikes again. I need to add a linter at some point to catch these kinds of things before merge. I'll get around to it at some point.

Thanks!

**Pull request merged**